sympy/physics/mechanics/joint.py

```diff
@@ -0,0 +1,356 @@
+        >>> J2 = PinJoint('J2', rod, plate, coordinates=phi, speeds=alpha, parent_joint_pos=lC, parent_axis=A.z, child_axis=B.z)
+
+    We can check the kinematics now.
+
+        >>> J1.kdes()
+        [omega(t) - Derivative(theta(t), t)]
+        >>> J2.kdes()
+        [alpha(t) - Derivative(phi(t), t)]
+        >>> rod.masscenter.vel(N)
+        (h/4 - lB/2)*omega(t)*A.x
+        >>> plate.masscenter.vel(N)
+        ((h/4 - lB/2)*omega(t) + (h/4 + lB/2)*omega(t))*A.x
+        >>> A.ang_vel_in(N)
+        omega(t)*N.y
+        >>> B.ang_vel_in(N)
+        omega(t)*N.y + alpha(t)*A.z
+        >>> A.dcm(N)
+        Matrix([
+        [cos(theta(t)), 0, -sin(theta(t))],
+        [            0, 1,              0],
+        [sin(theta(t)), 0,  cos(theta(t))]])
+        >>> B.dcm(N)
+        Matrix([
+        [ cos(phi(t))*cos(theta(t)), sin(phi(t)), -sin(theta(t))*cos(phi(t))],
+        [-sin(phi(t))*cos(theta(t)), cos(phi(t)),  sin(phi(t))*sin(theta(t))],
+        [             sin(theta(t)),           0,              cos(theta(t))]])
+
+    This example can also be done in another way, i.e, without explicitly defining constants,
+    dyamicsymbols, frames etc.
+
+        >>> rod = Body('rod')
+        >>> plate = Body('plate')
+        >>> C = Body('C') #Ceiling
+        >>> lA = (lB - h / 2) / 2 * rod.frame.z
+        >>> lC = (lB/2 + h/4) * rod.frame.z
+        >>> J1 = PinJoint('J1', C, rod, child_joint_pos=lA, parent_axis=C.frame.y, child_axis=rod.frame.y)
+        >>> J2 = PinJoint('J2', rod, plate, parent_joint_pos=lC, parent_axis=rod.frame.z, child_axis=plate.frame.z)
+        >>> J1.kdes()
+        [omega_J1(t) - Derivative(theta_J1(t), t)]
+        >>> J2.kdes()
+        [omega_J2(t) - Derivative(theta_J2(t), t)]
+        >>> rod.masscenter.vel(C.frame)
+        (h/4 - lB/2)*omega_J1(t)*rod_frame.x
+        >>> plate.masscenter.vel(C.frame)
+        ((h/4 - lB/2)*omega_J1(t) + (h/4 + lB/2)*omega_J1(t))*rod_frame.x
+        >>> rod.frame.ang_vel_in(C.frame)
+        omega_J1(t)*C_frame.y
+        >>> plate.frame.ang_vel_in(C.frame)
+        omega_J2(t)*rod_frame.z + omega_J1(t)*C_frame.y
+        >>> rod.frame.dcm(C.frame)
+        Matrix([
+        [cos(theta_J1(t)), 0, -sin(theta_J1(t))],
+        [               0, 1,                 0],
+        [sin(theta_J1(t)), 0,  cos(theta_J1(t))]])
+        >>> plate.frame.dcm(C.frame)
+        Matrix([
+        [ cos(theta_J1(t))*cos(theta_J2(t)), sin(theta_J2(t)), -sin(theta_J1(t))*cos(theta_J2(t))],
+        [-sin(theta_J2(t))*cos(theta_J1(t)), cos(theta_J2(t)),  sin(theta_J1(t))*sin(theta_J2(t))],
+        [                  sin(theta_J1(t)),                0,                   cos(theta_J1(t))]])
+
+    Notes
+    ======
+
+    All dynamicsymbols used in `Joints` are `real ` by default. It is adviced to use
+    `real=True` assumption when passing dynamicsymbols to `Joints` else it may give
+    incorrect result in some cases.
```

Comment:
We should be able to handle this without the user having to worry about this. We know this is a pin joint and we know what the resulting kinematics should be, so internally we should handle this and just give the correct answers without requiring this assumption to be set on variables.

Reply:
I still think we should set this assumptions, and make users habitual of using certain assumptions with mechanics. It's a new class so we can start it from here.

Reply:
I'd agree if we weren't dealing with these joints where we know what the answer should be already. It's useful for the developer of the joints to know this and use it to get the correct answer, but for users of Joint classes, we can simply give them the correct answer for the joint.

Reply:
So what do you suggest? How can we deal with this.